File: Java/src/main/java/parrot/NorwegianBlueParrot.java
package parrot;

public class NorwegianBlueParrot extends Parrot {
    private double voltage;
    private boolean isNailed;

    public NorwegianBlueParrot(ParrotTypeEnum type, int numberOfCoconuts, double voltage, boolean isNailed) {
        super(type, numberOfCoconuts, voltage, isNailed);
        this.voltage = voltage;
        this.isNailed = isNailed;
    }

    @Override
    public double getSpeed() {
        return super.getSpeed();
    }
}

File: Kotlin/src/main/kotlin/parrot/NorwegianBlueParrot.kt
package parrot

import kotlin.math.min

class NorwegianBlueParrot(
    voltage: Double,
    private val isNailed: Boolean
) : Parrot(
    voltage = voltage
) {

    private val nailedSpeed: Double
        get() = 0.0

    override val speed: Double
        get() = if (isNailed) nailedSpeed else getBaseSpeed(voltage)


    private val minBaseSpeed: Double
        get() = 24.0

    protected fun getBaseSpeed(voltage: Double): Double = min(minBaseSpeed, voltage * baseSpeed)
}
